src/gluonnlp/data/sampler.py

```diff
@@ -289,12 +288,16 @@ class BoundedBudgetSampler(BaseSampler):
         Number of partitions which the data is split into (default: 1)
     part_index
         The index of the part to read from
+    even_size
+        If the number of batches is not even across all partitions, sample a few extra batches
+        for the ones with fewer batches.
     """
     def __init__(self, lengths: Union[Sequence[int], Sequence[Sequence[int]]],
                  max_num_tokens: int = -1, max_num_sentences: int = -1,
                  required_batch_size_multiple: int = 1,
                  shuffle: bool = True, seed: Optional[int] = None,
-                 num_parts: int = 1, part_index: int = 0):
+                 num_parts: int = 1, part_index: int = 0,
+                 even_size: bool = False):
```

Comment:
I suggest wrapping the handling of `even_size` and `num_parts` into another function.

Reply:
Is it possible to write something similar to https://github.com/NVIDIA/DeepLearningExamples/blob/d875531dd69de92c8424e2254df6f393088e7ef4/PyTorch/Translation/Transformer/fairseq/data/data_utils.py#L65-L89?

Reply:
changed to ShardedIterator class in sampler.py